src/BenchmarkDotNet/Jobs/JobExtensions.cs

```diff
@@ -184,6 +184,31 @@ public static class JobExtensions
             job.WithCore(j => j.Environment.EnvironmentVariables = environmentVariables);
 
         public static Job With(this Job job, IReadOnlyList<Argument> arguments) => job.WithCore(j => j.Infrastructure.Arguments = arguments);
+        
+        /// <summary>
+        /// Runs the job with a specific Nuget dependency which will be resolved during the Job build process
+        /// </summary>
+        /// <param name="job"></param>
+        /// <param name="packageName">The Nuget package name</param>
+        /// <param name="packageVersion">The Nuget package version</param>
+        /// <returns></returns>
+        public static Job WithNuget(this Job job, string packageName, string packageVersion) => job.WithCore(j => j.Infrastructure.NugetReferences = new List<NugetReference> { new NugetReference(packageName, packageVersion) });
```

Comment:
the current value of `j.Infrastructure.NugetReferences`  is replaced with the new package, not added to the list.

```
job
   .WithNuget("A". "1.0.0");
   .WithNuget("B". "1.0.0");
```

would result only with a list of single element (B)